### DISCUSSION TITLE
[Question] Do you know if anyone is trying to write a SQL optimizer with egg?

### DISCUSSION THREAD

**Discussion by fabianmurariu** [egraphs-good/egg#254] [Q&A] [answered]
I'm considering trying myself to implement a Cascades/[Columbia](http://web.cecs.pdx.edu/~len/Columbia/#ref) top down optimizer using Egg and was wondering if anyone else was already trying.

**Comment by mwillsey** [comment, marked as answer]
I think the top-down aspect seem challenging, as egg is very bottom-up at the moment. I think this would be a good direction for future research. As far as SQL optimizers, you might take a look at what the [RisingLightDB folks are doing with egg](https://rustmagazine.org/issue-2/write-a-sql-optimizer-using-egg/).
